### ISSUE TITLE
OpenCV Exception

### ISSUE THREAD

**Issue by COLINDBB** [DennisLiu1993/Fastest_Image_Pattern_Matching#54]
Hello! 

I'm seeing an exception thrown in **matchTemplate**  
`what():  OpenCV(4.1.1-dev) /opencv/modules/imgproc/src/templmatch.cpp:1109: error: (-215:Assertion failed) _img.size().height <= _templ.size().height && _img.size().width <= _templ.size().width in function 'matchTemplate'`

Do you know what the issue may be here? 
These are my input images. 

![dst](https://github.com/DennisLiu1993/Fastest_Image_Pattern_Matching/assets/27859402/967991f8-a150-478a-9472-85706c8bc069)
![src](https://github.com/DennisLiu1993/Fastest_Image_Pattern_Matching/assets/27859402/015ce402-1a3d-410a-a784-16ed4fd2458c)

**Comment by DennisLiu1993** [comment]
template image should be smaller than source no matter in width or height

**Comment by Duy-Phuong-20** [comment]
@DennisLiu1993: can you tell me how can I run this project in python code as your demo because I am not familiar with C++??

**Comment by DennisLiu1993** [comment]
> @DennisLiu1993: can you tell me how can I run this project in python code as your demo because I am not familiar with C++??

extract pure c++ part to a class, and built it with an interface then it can be imported by python using ctype or pybind11

**Comment by Duy-Phuong-20** [comment]
> > @DennisLiu1993: can you tell me how can I run this project in python code as your demo because I am not familiar with C++??
> 
> extract pure c++ part to a class, and built it with an interface then it can be imported by python using ctype or pybind11

Yeah, thanks so much for your reply. I spent hours to read file .cpp and .h in matchtool folder. Can you show me which library .cpp and .h is necessary? Basically, I just need  the code that I input an image and take the result. So I tried to find the relationship between your libraries, but it is really difficult since there are so much code !!

**Comment by DennisLiu1993** [comment]
just focus on matchToolDlg.cpp/.h
